blockly-core/core/connections/connection.js

```diff
@@ -542,7 +542,8 @@ Blockly.Connection.prototype.closest = function(maxLimit, dx, dy) {
     var targetSourceBlock = connection.sourceBlock_;
 
     // Don't offer to connect to hidden blocks, unless we're in edit mode
-    if (!Blockly.editBlocks && !targetSourceBlock.isUserVisible()) {
+    if (!Blockly.editBlocks && (!targetSourceBlock.isUserVisible() ||
+        targetSourceBlock.isCurrentlyHidden())) {
```

Comment:
This could be a nice helper on block, e.g., `Block.prototype.shouldBeVisible = function () { (!targetSourceBlock.isUserVisible() || targetSourceBlock.isCurrentlyHidden() }` to avoid the behaviors diverging by external usages